support-frontend/assets/helpers/subscriptionsForms/__tests__/validationTest.ts

```diff
@@ -69,6 +70,65 @@ describe('validation', () => {
 		});
 	});
 
+	describe('requiredDeliveryAgentChosen', () => {
+		it('should return true if the fulfilment option is not HomeDelivery', () => {
+			expect(
+				requiredDeliveryAgentChosen('Collection', { isLoading: false }),
+			).toBe(true);
+		});
+
+		it('should return true if the fulfilment option is HomeDelivery and no agents are available', () => {
+			expect(
+				requiredDeliveryAgentChosen('HomeDelivery', { isLoading: false }),
+			).toBe(true);
+		});
```

Comment:
This case was a bit unexpected. I guess we don’t want this error to fire in this situation, because we’ll have another error that prevents the user from continuing?

I suppose this can happen when we actually want the user to continue with our standard within-M25 home delivery, as well as when they’re outside the M25 and we can’t deliver to their address, and we only want to stop them in the latter case.

Reply:
yes, exactly - we want this to fire only when the choice is available but they haven't chosen. Could the naming be more clear that this is what the validation is doing?

Reply:
I guess you could call it `deliveryAgentChosenIfRequiredAndAvailable` 🫣 I think it’s probably fine as-is